src/neo4j_genai/retrievers/base.py

```diff
@@ -75,24 +73,63 @@ def _fetch_index_infos(self):
         self._embedding_node_property = result["properties"][0]
         self._embedding_dimension = result["dimensions"]
 
+    def search(self, *args, **kwargs) -> RetrieverResult:
+        """
+        Search method. Call the get_search_result method that returns
+        a list of neo4j.Record, and format them to return RetrieverResult.
+        """
+        raw_result = self.get_search_results(*args, **kwargs)
+        formatter = self.get_result_formatter()
+        search_items = [
+            formatter(record)
+            for record in raw_result.records
+        ]
+        metadata = raw_result.metadata or {}
+        metadata["__retriever"] = self.__class__.__name__
+        return RetrieverResult(
+            items=search_items,
+            metadata=metadata,
+        )
+
+    @abstractmethod
+    def get_search_results(self, *args, **kwargs) -> RetrieverRawResult:
```

Comment:
If only `.search()` is exposed to users, should we make this a private method `_get_search_results()`?